sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareDirectoryAsyncClient.java

```diff
@@ -605,18 +609,50 @@ private boolean checkDoesNotExistStatusCode(Throwable t) {
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedFlux<ShareFileItem> listFilesAndDirectories(String prefix, Integer maxResultsPerPage) {
         try {
-            return listFilesAndDirectoriesWithOptionalTimeout(prefix, maxResultsPerPage, null, Context.NONE);
+            return listFilesAndDirectoriesWithOptionalTimeout(new ShareDirectoryListFilesAndDirectoriesOptions()
+                .setPrefix(prefix).setMaxResultsPerPage(maxResultsPerPage), null, Context.NONE);
         } catch (RuntimeException ex) {
             return pagedFluxError(logger, ex);
         }
     }
 
-    PagedFlux<ShareFileItem> listFilesAndDirectoriesWithOptionalTimeout(String prefix, Integer maxResultsPerPage,
-                                                                        Duration timeout, Context context) {
+    /**
+     * Lists all sub-directories and files in this directory with their prefix or snapshots.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * <p>List all sub-directories with "subdir" prefix and return 10 results in the account</p>
+     *
+     * {@codesnippet com.azure.storage.file.share.ShareDirectoryAsyncClient.listFilesAndDirectories#ShareDirectoryListFilesAndDirectoriesOptions}
+     *
+     * <p>For more information, see the
+     * <a href="https://docs.microsoft.com/rest/api/storageservices/list-directories-and-files">Azure
+     * Docs</a>.</p>
+     *
+     * @param options Optional parameters.
+     * the server will return up to 5,000 items.
+     * @return {@link ShareFileItem File info} in this directory with prefix and max number of return results.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<ShareFileItem> listFilesAndDirectories(ShareDirectoryListFilesAndDirectoriesOptions options) {
+        try {
+            return listFilesAndDirectoriesWithOptionalTimeout(options, null, Context.NONE);
+        } catch (RuntimeException ex) {
+            return pagedFluxError(logger, ex);
+        }
+    }
+
+    PagedFlux<ShareFileItem> listFilesAndDirectoriesWithOptionalTimeout(
+        ShareDirectoryListFilesAndDirectoriesOptions options, Duration timeout, Context context) {
+        StorageImplUtils.assertNotNull("options", options);
```

Comment:
If this options bag has no required params, theres no need to assertNotNull, you can just do `options = options == null ? new ShareDirectoryListFilesAndDirectoriesOptions() : options;`